sdk/eventgrid/Microsoft.Azure.CloudNative.CloudEvents.EventGrid/tests/CloudEventTests.cs

```diff
@@ -0,0 +1,137 @@
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+using Azure.Core.TestFramework;
+using Azure.Messaging.EventGrid;
+using CloudNative.CloudEvents;
+using NUnit.Framework;
+using CloudEvent = CloudNative.CloudEvents.CloudEvent;
+
+namespace Microsoft.Azure.CloudNative.CloudEvents.EventGrid.Tests
+{
+    public class CloudEventTests
+    {
+
+        private const string TraceParentHeaderName = "traceparent";
+        private const string TraceStateHeaderName = "tracestate";
+
+        private static readonly JsonEventFormatter s_eventFormatter = new JsonEventFormatter();
+
+
+        [Test]
+        [TestCase(false, false)]
+        [TestCase(true, true)]
+        [TestCase(true, false)]
+        [TestCase(false, true)]
+        public async Task SetsTraceParentExtension(bool inclTraceparent, bool inclTracestate)
+        {
+            var mockTransport = new MockTransport(new MockResponse(200));
+            var options = new EventGridPublisherClientOptions
+            {
+                Transport = mockTransport
+            };
+            EventGridPublisherClient client =
+               new EventGridPublisherClient(
+                   new Uri("http://localHost"),
+                   new AzureKeyCredential("fakeKey"),
+                   options);
+            var activity = new Activity($"{nameof(EventGridPublisherClient)}.{nameof(EventGridPublisherClient.SendEvents)}");
+            activity.SetW3CFormat();
+            activity.Start();
+            List<CloudEvent> eventsList = new List<CloudEvent>();
+            for (int i = 0; i < 10; i++)
+            {
+                var cloudEvent =
+                    new CloudEvent(
+                        "record",
+                        new Uri("http://localHost"),
+                        Guid.NewGuid().ToString(),
+                        DateTime.Now);
+
+                if (inclTraceparent && inclTracestate && i % 2 == 0)
+                {
+                    cloudEvent.GetAttributes().Add("traceparent", "traceparentValue");
+                }
+                if (inclTracestate && i % 2 == 0)
+                {
+                    cloudEvent.GetAttributes().Add("tracestate", "param:value");
+                }
+                eventsList.Add(cloudEvent);
+            }
+            await client.SendCloudEventsAsync(eventsList);
+
+            activity.Stop();
+            List<CloudEvent> cloudEvents = DeserializeRequest(mockTransport.SingleRequest);
+            IEnumerator<CloudEvent> cloudEnum = eventsList.GetEnumerator();
+            foreach (CloudEvent cloudEvent in cloudEvents)
+            {
+                cloudEnum.MoveNext();
+                IDictionary<string, object> cloudEventAttr = cloudEnum.Current.GetAttributes();
+                if (cloudEventAttr.ContainsKey(TraceParentHeaderName) &&
```

Comment:
use `inclTraceparent` and `inclTracestate` for conditions here.

Reply:
I have that i%2 check so it is not enough to check those flags.